packages/frontend/src/features/sqlRunner/store/tableVisSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { ChartKind, type TableChartSqlConfig } from '@lightdash/common';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { createSlice } from '@reduxjs/toolkit';
+import { setInitialResultsAndSeries, setSaveChartData } from './sqlRunnerSlice';
+
+const initialState: { config: TableChartSqlConfig | undefined } = {
+    config: undefined,
+};
+
+export const tableVisSlice = createSlice({
+    name: 'tableVisConfig',
+    initialState,
+    reducers: {
+        updateFieldLabel: (
+            { config },
+            action: PayloadAction<Record<'reference' | 'label', string>>,
+        ) => {
+            const { reference, label } = action.payload;
+            if (config && config.columns[reference]) {
+                config.columns[reference].label = label;
+            }
+        },
+    },
+    extraReducers: (builder) => {
+        builder.addCase(setInitialResultsAndSeries, (state, action) => {
+            if (state.config === undefined) {
+                // TODO: this should come from the transformer
+                const columns = Object.keys(action.payload[0]).reduce<
```

Comment:
This is duplicated per table and should go on the transformer, but we are still working out the design for transformers. We can move this when that's ready.